test/unit/range/view/view_pairwise_combine_test.cpp

```diff
@@ -0,0 +1,511 @@
+{
+    using iterator_t = std::ranges::iterator_t<typename TestFixture::view_t>;
+
+    auto r = this->resource();
+    iterator_t it{r.begin(), r.begin(), r.end()};
+
+    if constexpr (std::RandomAccessIterator<decltype(r.begin())>)
+    {
+        it = it + 1;
+        EXPECT_EQ(*it, (std::tuple{'a', 'c'}));
+        it += 2;
+        EXPECT_EQ(*it, (std::tuple{'b', 'c'}));
+        it = 2 + it;
+        EXPECT_EQ(*it, (std::tuple{'c', 'd'}));
+    }
+}
+
+TYPED_TEST(pairwise_combine_iterator_test, decrement_n)
+{
+    if constexpr (std::ranges::RandomAccessRange<TypeParam>)
+    {
+        using iterator_t = std::ranges::iterator_t<typename TestFixture::view_t>;
+
+        auto r = this->resource();
+        iterator_t it{std::ranges::prev(r.end()), r.begin(), r.end()};
+
+        it = it - 2;
+        EXPECT_EQ(*it, (std::tuple{'b', 'd'}));
+        it -= 3;
+        EXPECT_EQ(*it, (std::tuple{'a', 'c'}));
+    }
+}
+
+TYPED_TEST(pairwise_combine_iterator_test, distance)
+{
+    if constexpr (std::ranges::RandomAccessRange<TypeParam>)
+    {
+        using iterator_t = std::ranges::iterator_t<typename TestFixture::view_t>;
+
+        auto r = this->resource();
+        iterator_t it_1{r.begin(), r.begin(), r.end()};
+        iterator_t it_2{std::ranges::prev(r.end()), r.begin(), r.end()};
+
+        EXPECT_EQ(it_2 - it_1, 6);
+        --it_2;
+        EXPECT_EQ(it_2 - it_1, 5);
+        ++it_1;
+        EXPECT_EQ(it_2 - it_1, 4);
+    }
+}
+
+TYPED_TEST(pairwise_combine_iterator_test, order)
+{
+    using iterator_t = std::ranges::iterator_t<typename TestFixture::view_t>;
+
+    auto r = this->resource();
+    iterator_t it_1{r.begin(), r.begin(), r.end()};
+    iterator_t it_2{r.begin(), r.begin(), r.end()};
+    ++it_2;
+
+    if constexpr (std::StrictTotallyOrdered<decltype(r.begin())>)
+    {
+        EXPECT_FALSE(it_1 < it_1);
+        EXPECT_TRUE(it_1 <= it_1);
+        EXPECT_FALSE(it_1 > it_1);
+        EXPECT_TRUE(it_1 >= it_1);
+
+        EXPECT_TRUE(it_1 < it_2);
+        EXPECT_TRUE(it_1 <= it_2);
+        EXPECT_FALSE(it_1 > it_2);
+        EXPECT_FALSE(it_1 >= it_2);
+    }
+}
+
+TYPED_TEST(pairwise_combine_test, view_concept)
+{
+    EXPECT_TRUE(std::ranges::InputRange<typename TestFixture::view_t>);
+    EXPECT_TRUE(std::ranges::ForwardRange<typename TestFixture::view_t>);
+    EXPECT_TRUE(std::ranges::View<typename TestFixture::view_t>);
+    EXPECT_FALSE((std::ranges::OutputRange<typename TestFixture::view_t, std::tuple<char &, char &>>));
+
+    if constexpr (std::ranges::BidirectionalRange<TypeParam>)
+    {
+        EXPECT_TRUE(std::ranges::BidirectionalRange<typename TestFixture::view_t>);
+    }
+
+    if constexpr (std::ranges::SizedRange<TypeParam>)
+    {
+        EXPECT_TRUE(std::ranges::SizedRange<typename TestFixture::view_t>);
+    }
+
+    if constexpr (std::ranges::RandomAccessRange<TypeParam>)
+    {
+        EXPECT_TRUE(std::ranges::RandomAccessRange<typename TestFixture::view_t>);
+    }
```

Comment:
Can you EXPECT_EQ here instead (equivalence)? It's a stronger check, because your's is just implication.

Reply:
:nail_care: